### PR TITLE
This PR rewrites to `io.tess.TOI` object using pydantic 

### PR DESCRIPTION
**Pull request by mwcraig**
Doing so has a couple of advantages:

+ The code is more compact
+ It is easy to serlialize to disk

Helps address, but does not complete fix the issue #188

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -101,6 +101,8 @@ include = [
         # Don't complain if tests don't hit assertions
         "raise AssertionError",
         "raise NotImplementedError",
+        # Don't complain about connection errors
+        "raise requests.ConnectionError",
         # Don't complain about script hooks
```
Comment (JuanCab) [inline pyproject.toml:106]: Is this to avoid errors during testing if the TESS servers are not available?

File: stellarphot/io/tess.py
Comment (JuanCab) [inline stellarphot/io/tess.py:1]: Nice!  I had to read through it a bit, but pydantic classes do make some things a lot easier, like having properties.  That said, I wonder, can you now set the values of those properties by calling them or is pydantic preventing that.  Not a huge concern, but one nice feature of `@property` is allowing you to make read-only attributes... at the pydantic ones read-only once set?

**Review by JuanCab — APPROVED** [review]
I approve this, but would appreciate feedback on the questions.

File: pyproject.toml (hunk 1 above)
Comment (mwcraig) [reply pyproject.toml:106]: Yes, and I wasn't sure how to test for that case...

File: stellarphot/io/tess.py
Comment (mwcraig) [reply stellarphot/io/tess.py:1]: You can now set them, but pydantic has a way of making them non-settable: https://docs.pydantic.dev/latest/concepts/models/#faux-immutability

Lmk if you want me to do that

Comment (mwcraig) [reply stellarphot/io/tess.py:1]: (merging for now, though, so I can open the next PR)

**Pull request merged**